### ISSUE TITLE
SimulatedLocationManager crash

### ISSUE THREAD

**Issue by macdrevx** [mapbox/mapbox-navigation-ios#1662]
**Mapbox Navigation SDK version:**
0.18.1

**Crash log:**

```
# Date: 2018-09-08T03:36:00Z
# OS Version: 11.4.1 (15G77)
# Device: iPad 6
# RAM Free: 12%
# Disk Free: 79.4%

#0. Crashed: com.apple.main-thread
0  libobjc.A.dylib                0x18425891c objc_msgSend + 28
1  MapboxCoreNavigation           0x103819f94 _T020MapboxCoreNavigation24SimulatedLocationManagerC4tick33_BAD5A1656A34E7A88BC59C7E80655DACLLyyF + 1376
2  MapboxCoreNavigation           0x10381a3b0 _T020MapboxCoreNavigation24SimulatedLocationManagerC013startUpdatingE0yyFToTm + 28
3  Foundation                     0x185ac04ac __NSFireDelayedPerform + 408
4  CoreFoundation                 0x185033aa8 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 28
5  CoreFoundation                 0x18503376c __CFRunLoopDoTimer + 864
6  CoreFoundation                 0x185033010 __CFRunLoopDoTimers + 248
7  CoreFoundation                 0x185030b60 __CFRunLoopRun + 2168
8  CoreFoundation                 0x184f50da8 CFRunLoopRunSpecific + 552
9  GraphicsServices               0x186f36020 GSEventRunModal + 100
10 UIKit                          0x18ef70758 UIApplicationMain + 236
11 OnDemandDriver                 0x102caaecc main (main.swift:25)
12 libdyld.dylib                  0x1849e1fc0 start + 4
```

**Observations**

Our app isn't using `SimulatedLocationManager` directly, but we do use `NavigationViewController`, which has a `RouteController`, which has a `TunnelIntersectionManager`, which has a `SimulatedLocationManager`. As far as I can tell from an inspection of the source code, that's the only place where one is being used.

**Comment by akitchen** [comment]
I think there could be a subtle and difficult-to-reproduce bug in the `tick()` method, which is using `perform(#selector...` after canceling previously scheduled performs and doing some work to set up the next scheduled `perform(#selector...`.

`perform(#selector...` sets up a scheduled timer, which is retained by the run loop. in between the call to cancel previous performs and the call to `perform(#selector...` it is possible that the object could be unretained and therefore deallocated. At least that's my read of this right now.

In other words, I think there could be a race condition inside `tick()`. Not sure how to reproduce the bug more reliably but I do think there's something here which needs to be refactored. I'll see if we have a public ticket and write one if not.

Thanks for letting us know!

**Comment by Dick-Barnes** [comment]
is there an update on this.  I can reproduce it every time.  It is in the tick().  I haven't run on a real device yet.  Is this an issue only when running in the simulator?

**Comment by JThramer** [comment]
Fixed in #1662 cc @Dick-Barnes @macdrevx

**Comment by Dick-Barnes** [comment]
thanks...updated my cocoa pods install and problem resolved.
